// @ngx-validate/core/esm2020/lib/directives/validation-target.directive.mjs
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
export class ValidationTargetDirective {
    constructor(vcRef) {
        this.vcRef = vcRef;
    }
}
ValidationTargetDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.7", ngImport: i0, type: ValidationTargetDirective, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Directive });
ValidationTargetDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.2.7", type: ValidationTargetDirective, selector: "[validationTarget]", exportAs: ["validationTarget"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.7", ngImport: i0, type: ValidationTargetDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[validationTarget]',
                    exportAs: 'validationTarget',
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; } });
